tests/test_inner.py

```diff
@@ -755,36 +790,51 @@ def test_workflow_docker_build_error():
     """
     This is a test for what happens when the docker build fails.
     """
-
+    flexmock(DockerfileParser, content='df_content')
     this_file = inspect.getfile(PreRaises)
     mock_docker()
     fake_builder = MockInsideBuilder(failed=True)
     flexmock(InsideBuilder).new_instances(fake_builder)
+    watch_pre  = Watcher()
+    watch_build = Watcher()
     watch_prepub = Watcher()
     watch_post = Watcher()
     watch_exit = Watcher()
 
     workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image',
+                                   prebuild_plugins=[{'name': 'pre_watched',
+                                                      'args': {
+                                                          'watcher': watch_pre
+                                                      }}],
+                                   buildstep_plugins=[{'name': 'build_watched',
+                                                       'args': {
+                                                       'watcher': watch_build
+                                                       }}],
                                    prepublish_plugins=[{'name': 'prepub_watched',
                                                         'args': {
                                                             'watcher': watch_prepub,
                                                         }}],
                                    postbuild_plugins=[{'name': 'post_watched',
                                                        'args': {
-                                                           'watcher': watch_post
+                                                       'watcher': watch_post
                                                        }}],
                                    exit_plugins=[{'name': 'exit_watched',
                                                   'args': {
                                                       'watcher': watch_exit
                                                   }}],
                                    plugin_files=[this_file])
 
-    assert workflow.build_docker_image().is_failed()
+    flexmock(watch_build).should_call('was_called').and_raise(Exception)
```

Comment:
This doesn't really test the code then - exception should be thrown from `inner.py`